addons/project/models/project.py

```diff
@@ -699,6 +699,49 @@ def unlink(self):
         analytic_accounts_to_delete.unlink()
         return result
 
+    @api.model
+    def search(self, domain, offset=0, limit=None, order=None, count=False):
+        """ Copy/paste from knowledge_article.search().
+            Allows to order by is_favorite even though this field is compute not stored.
+        """
+        if count or not order or 'is_favorite' not in order:
+            return super(Project, self).search(domain, offset=offset, limit=limit, order=order, count=count)
+        order_items = [order_item.strip().lower() for order_item in (order or self._order).split(',')]
+        favorite_desc = any('is_favorite desc' in item for item in order_items)
+
+        # Search projects that are favorite of the current user.
+        new_domain = expression.AND([[('is_favorite', '=', True)], domain])
+        new_order = ', '.join(item for item in order_items if 'is_favorite' not in item)
+        project_ids = super().search(new_domain, offset=0, limit=None, order=new_order, count=count).ids
```

Comment:
Maybe we could use the limit to avoid fetching too much records? 🤔 
What do you think? 🙂 

```suggestion
        project_ids = super().search(new_domain, offset=0, limit=limit, order=new_order, count=count).ids
```